### PR TITLE
support maskedpath and readonlyPath

### PR DESCRIPTION
**Pull request by bergwolf**
fixes https://github.com/hyperhq/runv/issues/552

### CODE REVIEW

File: cli/filesystem.go [hunk 1]
```diff
@@ -158,6 +158,11 @@ func setupContainerFs(vm *hypervisor.Vm, bundle, container string, spec *specs.S
 		glog.Errorf("mount %s to %s failed: %s\n", rootPath, vmRootfs, err.Error())
 		return err
 	}
+	defer func() {
+		if err != nil {
+			utils.Umount(containerSharedFs)
+		}
+	}()
```
Comment (laijs) [inline cli/filesystem.go]: it seams it should be moved up before utils.Mount(rootPath, vmRootfs).

Comment (bergwolf) [inline cli/filesystem.go]: good point!

**Comment by bergwolf** [comment]
pr updated to address @laijs's comment and one more commit was added to fix ci failure.

**Comment by laijs** [comment]
Could you also enable the tests tests/bats-integration/mask.bats?
(modify it when needed)

File: Makefile.am [hunk 2]
```diff
@@ -18,13 +18,15 @@ HYPER_BULD_TAGS=$(XEN_BUILD_TAG) $(LIBVIRT_BUILD_TAG)
...
 clean-local:
 	-rm -f runv
-install-exec-local: 
+install-exec-local:
 	$(INSTALL_PROGRAM) runv $(bindir)
 
 build-runv:
 	go build -tags "static_build $(HYPER_BULD_TAGS)" -ldflags ${GOLDFLAGS} -o runv ./cli/
-test-integration:
+integration-fast:
```
Comment (laijs) [inline Makefile.am]: just use go-integration

File: tests/go-integration/test_data/Makefile [hunk 3]
```diff
@@ -36,6 +36,7 @@ busybox.tar: build/busybox
 	touch build/rootfs/.dockerenv
 	install -D build/busybox build/rootfs/bin/busybox
 	set -e; for i in `build/busybox --list`; do ln -s busybox build/rootfs/bin/$$i; done
+	mkdir -p build/rootfs/testdata && cp build/busybox build/rootfs/testdata/foobar
```
Comment (laijs) [inline tests/go-integration/test_data/Makefile]: build/rootfs/masktest/

Comment (bergwolf) [reply tests/go-integration/test_data/Makefile]: It's used in two tests, maskedpath and readonlypath.

**Comment by bergwolf** [comment]
pr updated and mask.bats enabled.

**Comment by laijs** [comment]
LGTM

**Pull request merged**